staging/src/k8s.io/apiserver/pkg/storage/cacher/time_budget.go

```diff
@@ -48,48 +54,38 @@ type timeBudgetImpl struct {
 	sync.Mutex
 	budget time.Duration
 
-	refresh   time.Duration
 	maxBudget time.Duration
+	clock     clock.Clock
+	limiter   *rate.Limiter
 }
 
-func newTimeBudget(stopCh <-chan struct{}) timeBudget {
+func newTimeBudget() timeBudget {
+	clock := clock.RealClock{}
+	limiter := rate.NewLimiter(rate.Limit(1), burstSize)
```

Comment:
Nice. But I would actually say that we should go one step further.

Instead of adding a unit every 1s, we should increase the frequency, and e.g. add an equivalent of 5ms every 100ms or sth like that.

Reply:
done, PTAL